code/solr/reindex/handlers/SolrReindexImmediateHandler.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+use Psr\Log\LoggerInterface;
+
+/**
+ * Invokes an immediate reindex
+ *
+ * Internally batches of records will be invoked via shell tasks in the background
+ */
+class SolrReindexImmediateHandler extends SolrReindexBase {
+
+	public function triggerReindex(LoggerInterface $logger, $batchSize, $taskName, $classes = null) {
+		$this->runReindex($logger, $batchSize, $taskName, $classes);
+	}
+
+	protected function processIndex(
+		LoggerInterface $logger, SolrIndex $indexInstance, $batchSize, $taskName, $classes = null
+	) {
+		parent::processIndex($logger, $indexInstance, $batchSize, $taskName, $classes);
+
+		// Immediate processor needs to immediately commit after each index
+		$indexInstance->getService()->commit();
+	}
+
+	/**
+	 * Process a single group.
+	 *
+	 * Without queuedjobs, it's necessary to shell this out to a background task as this is
+	 * very memory intensive.
+	 *
+	 * The sub-process will then invoke $processor->runGroup() in {@see Solr_Reindex::doReindex}
+	 *
+	 * @param LoggerInterface $logger
+	 * @param SolrIndex $indexInstance Index instance
+	 * @param array $state Variant state
+	 * @param string $class Class to index
+	 * @param int $groups Total groups
+	 * @param int $group Index of group to process
+	 * @param string $taskName Name of task script to run
+	 */
+	protected function processGroup(
+		LoggerInterface $logger, SolrIndex $indexInstance, $state, $class, $groups, $group, $taskName
+	) {
+		// Build state
+		$statevar = json_encode($state);
+		if (strpos(PHP_OS, "WIN") !== false) {
+			$statevar = '"'.str_replace('"', '\\"', $statevar).'"';
+		} else {
+			$statevar = "'".$statevar."'";
+		}
+
+		// Build script
+		$indexName = $indexInstance->getIndexName();
+		$scriptPath = sprintf("%s%sframework%scli-script.php", BASE_PATH, DIRECTORY_SEPARATOR, DIRECTORY_SEPARATOR);
+		$scriptTask = "php {$scriptPath} dev/tasks/{$taskName}";
```

Comment:
This is what the code was using before, so I guess this just keeps the same resolution mechanism in place.

Although in the past the php reference was in Solr_Reindex class.